src/main/java/org/spongepowered/api/data/DataRegistration.java

```diff
@@ -40,8 +41,12 @@
      * the desired {@link DataManipulator} type to be registered.
      *
      * @return The new builder instance
+     * @deprecated Use the {@link DataGenerator} instead, this is the new
+     *    method of generating {@link DataManipulator} implementations and
```

Comment:
Maybe not specifically in implementation, but I don't see why the actual builder method is being removed if it still can be used, whether developers choose to implement custom data manipulators themselves, or using dynamic class generation through the API.